### PR TITLE
[EM-0/W-1] Initialize Next.js 14 project with TypeScript and package.js

### PR DESCRIPTION
**Pull request by mohsen1**
## Task
Initialize Next.js 14 project with TypeScript and package.json configuration

## Files Changed
- package.json
- package-lock.json
- tsconfig.json
- next.config.js
- .gitignore
- .nvmrc

---
*Automated by Claude Code Orchestrator*

### CODE REVIEW

File: db/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Database from "better-sqlite3";
+import { drizzle } from "drizzle-orm/better-sqlite3";
+import * as schema from "./schema";
+
+// Create database directory if it doesn't exist
+const fs = require("fs");
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply db/index.ts]: Fixed. Replace 'const fs = require("fs");' with 'import fs from "fs";' at the top of the file, maintaining the ES module import style used throughout the rest of the TypeScript codebase.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      }
+    ],
+    "@typescript-eslint/no-explicit-any": "error",
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/explicit-module-boundary-types": "off",
+    "@typescript-eslint/no-non-null-assertion": "warn",
+    "no-console": [
+      "warn",
+      {
+        "allow": ["warn", "error"]
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply .eslintrc.json]: Fixed. Update .eslintrc.json line 23 to change the no-console allow list from ['warn', 'error'] to ['warn', 'error', 'log'] to match the actual usage in server.ts. Alternatively, if the intention is to restrict console usage, update server.ts to replace console.log calls with console.warn or console.error as appropriate.

File: server.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  });
+
+  io.on("connection", (socket) => {
+    console.log("Client connected:", socket.id);
+
+    // Join a group room
+    socket.on("join_group", ({ groupId, userId }) => {
+      // TODO: Validate user has access to this group
+      socket.join(groupId);
+      console.log(`User ${userId} joined group ${groupId}`);
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply server.ts]: Fixed. Implement the suggested code which: 1) Extracts authenticated userId from socket handshake (auth or query), 2) Validates that the claimed userId matches the authenticated userId, 3) Performs basic groupId validation, 4) Checks against allowedGroups array in handshake.auth to verify group membership authorization, 5) Returns early with error message if any validation fails before allowing the socket to join the group.

File: vitest.setup.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import { drizzle } from "drizzle-orm/better-sqlite3";
+import * as schema from "./db/schema";
+
+// Setup in-memory database for testing
+let db: ReturnType<typeof drizzle>;
+
+beforeAll(() => {
+  const sqlite = new Database(":memory:");
+  sqlite.pragma("foreign_keys = ON");
+  db = drizzle(sqlite, { schema });
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply vitest.setup.ts]: Fixed. Add an export statement for the db variable. Change the declaration from 'const db = ...' to 'export const db = ...' or add 'export { db }' after the declaration line.

File: lib/utils.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  const distribution = Array(numPeople).fill(baseAmount);
+
+  // Distribute remainder pennies to the first 'remainder' people
+  for (let i = 0; i < remainder; i++) {
+    distribution[i] += 1;
+  }
+
+  return distribution;
+}
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply lib/utils.ts]: Fixed. Create a new test file (e.g., lib/utils.test.ts or similar) that adds comprehensive test coverage for formatCurrency, toCents, and distributeEvenly functions. Tests should include: edge cases (zero, negative values, decimal precision), typical use cases, and specific tests for distributeEvenly to verify correct remainder distribution across multiple recipients.

File: drizzle.config.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import type { Config } from 'drizzle-kit';
+
+export default {
+  schema: './db/schema.ts',
+  out: './db/migrations',
+  driver: 'better-sqlite',
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply drizzle.config.ts]: Fixed. Replace 'driver: "sqlite"' with 'dialect: "sqlite"' in the drizzle.config.ts file at line 6

File: server.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mohsen1) [reply server.ts]: Fixed. Implement the suggested authorization checks in the join_group handler: 1) Extract and validate the authenticated userId from the handshake, 2) Verify that the claimed userId matches the authenticated userId, 3) Validate the groupId format, 4) If allowed groups are specified in the handshake auth, enforce membership by checking if the requested groupId is in the allowed groups list. This prevents unauthorized users from joining groups they shouldn't have access to.

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        with:
+          node-version: 20
+          cache: "npm"
+      - run: npm ci
+
+      # 4. Build Verification (Ensures Next.js build passes)
+      - name: Next Build
+        run: npm run build
+        env:
+          NEXT_PUBLIC_API_URL: "http://localhost:3000"
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply .github/workflows/ci.yml]: Fixed. Change line 54 in .github/workflows/ci.yml from 'NEXT_PUBLIC_API_URL: "http://localhost:3000"' to 'NEXT_PUBLIC_APP_URL: "http://localhost:3000"' to match the environment variable name used in .env.example and next.config.js

File: server.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    path: "/api/socket",
+    addTrailingSlash: false,
+  });
+
+  // Socket.IO authentication middleware
+  io.use((socket, next) => {
+    // In production, you would validate the session token here
+    // For now, we'll allow all connections
+    next();
+  });
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply server.ts]: Fixed. Implement proper authentication in the Socket.IO middleware. The middleware should verify the user's JWT token or session from the socket handshake headers/query parameters before allowing the connection. Example: extract and verify a JWT token from socket.handshake.auth.token or socket.handshake.headers.authorization, reject the connection if invalid, and attach the authenticated user object to the socket for use in subsequent event handlers.

File: lib/utils.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (mohsen1) [reply lib/utils.ts]: Fixed. Create a test file (e.g., lib/utils.test.ts) and add comprehensive unit tests for:
1. formatCurrency - test various currency amounts, edge cases (zero, negative, large numbers)
2. toCents - test conversion from dollars to cents, including decimal handling
3. distributeEvenly - test even distribution of amounts, remainder handling, edge cases (zero, insufficient funds, rounding scenarios)

Run tests with vitest to ensure they pass and cover edge cases.

File: drizzle.config.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (mohsen1) [reply drizzle.config.ts]: Fixed. Replace the 'driver: "sqlite"' configuration option with 'dialect: "sqlite"' in the drizzle.config.ts file at line 6. This is a straightforward rename of the configuration property that aligns with the current drizzle-kit API.

File: .github/workflows/ci.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (mohsen1) [reply .github/workflows/ci.yml]: Fixed. Change line 54 in .github/workflows/ci.yml from 'NEXT_PUBLIC_API_URL: "http://localhost:3000"' to 'NEXT_PUBLIC_APP_URL: "http://localhost:3000"' to match the environment variable name used throughout the application.

File: vitest.setup.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import { beforeAll, afterEach, afterAll } from "vitest";
```
[the remark this answers is not in the document]
Comment (mohsen1) [reply vitest.setup.ts]: Fixed. Remove the unused 'afterEach' import from the import statement, keeping only 'beforeAll' and 'afterAll' that are actually being used in the file.
